frontend/src/store/authn/helper.js

```diff
@@ -61,13 +60,17 @@ function isExpired (user) {
 export function useUserManager (options) {
   const {
     cookies = useCookies([COOKIE_HEADER_PAYLOAD]),
-    location = useBrowserLocation(),
     logger = useLogger(),
   } = options ?? {}
 
   let refreshTokenPromise
+  let signoutInProgress = false
 
-  const origin = unref(location).origin
+  useEventListener(window, 'beforeunload', () => {
+    signoutInProgress = true
+  })
```

Comment:
for which case is this relevant?

Reply:
This is not strictly necessary, especially if the page is going to be completely unloaded during the sign-out process. Using `beforeunload` is more about keeping the state clean in real-time, which might not offer any practical benefit in this case.